internal/engine/configs/configs_controller.go

```diff
@@ -87,9 +87,10 @@ func (cc *ConfigsController) loadTiltfile(ctx context.Context, st store.RStore,
 	if !firstBuild {
 		logTiltfileChanges(ctx, filesChanged)
 	}
+	flagsState := state.FlagsState
 	st.RUnlockState()
 
-	tlr := cc.tfl.Load(ctx, tiltfilePath, args)
+	tlr := cc.tfl.Load(ctx, tiltfilePath, args, flagsState)
```

Comment:
should FlagsState include `args`?

it feels like you always have to use them together, because `FlagsState` is more like "metadata about where the args came from"

Reply:
+1

Reply:
Yep, I actually already ended up doing this in the next PR. If it's ok, I won't spend the effort to pull that change into this PR.